assets/js/components/GoogleChart/index.js

```diff
@@ -441,30 +439,11 @@ export default function GoogleChart( props ) {
 				{ !! dateMarkersInRange.length &&
 					dateMarkersInRange.map( ( dateMarker, index ) => {
 						return (
-							<Fragment
+							<DateMarkers
 								key={ `googlesitekit-chart__date-marker--${ instanceID }-${ index }` }
-							>
-								<div
-									id={ `googlesitekit-chart__date-marker-line--${ instanceID }-${ index }` }
-									className="googlesitekit-chart__date-marker-line"
-								/>
-								{ dateMarker.text && (
-									<div
-										id={ `googlesitekit-chart__date-marker-tooltip--${ instanceID }-${ index }` }
-										className="googlesitekit-chart__date-marker-tooltip"
-									>
-										<Tooltip title={ dateMarker.text }>
-											<span>
-												<Icon
-													fill="currentColor"
-													icon={ info }
-													size={ iconSize }
-												/>
-											</span>
-										</Tooltip>
-									</div>
-								) }
-							</Fragment>
+								id={ `${ instanceID }-${ index }` }
+								dateMarker={ dateMarker }
```

Comment:
Since this component is only responsible for a single marker, let's rename it to be singular, and then pass the `dateMarker` through as props. Looks like it only uses `text` internally for now so we could only pass that through but we could also pass the `date` for completeness.

In the future it would be nice to extract more date marker specific code into a hook or another component but this is a good start.

Reply:
Updated, thank you!